api/app/jobs/model_data.md

```diff
@@ -0,0 +1,122 @@
+- [Formatting of filenames](https://weather.gc.ca/grib/grib2_reg_10km_e.html)
+- [Data download source](https://dd.weather.gc.ca/model_gem_regional/10km/grib2/)
+
+##### HRDPS
+- [HRDPS weather variables for 0-hour forecasts](https://weather.gc.ca/grib/HRDPS_HR/HRDPS_nat_ps2p5km_P000_deterministic_e.html)
+- [HRDPS weather variables for non-zero-hour forecasts](https://weather.gc.ca/grib/HRDPS_HR/HRDPS_ps2p5km_PNONZERO_deterministic_e.html)
+- [Formatting of filenames](https://weather.gc.ca/grib/grib2_HRDPS_HR_e.html)
+- [Data download source](https://dd.weather.gc.ca/model_hrdps/continental/grib2/)
+
+#### Relevant Weather Variables
+
+The GDPS, RDPS, and HRDPS weather variables that we currently fetch for use in Morecast are `TMP_TGL_2m` (temperature), `RELH_TGL_2m` (relative humidity), `WIND_TGL_10m` (wind speed), `WDIR_TGL_10m` (wind direction), and `APCP_SFC_0` (accumulated precipitation).
+
+### NOAA
+
+We fetch [GFS model data from NCEP-NOAA](https://www.ncei.noaa.gov/products/weather-climate-models/global-forecast). NOAA outputs only one GRIB file for all weather variables for a specific timestamp, so each of their GRIB files contains hundreds of raster bands, with one raster band for each weather variable. We therefore must download the entire GRIB file, even though we only need data from a small number of the raster bands in the file.
+
+Specifically, we fetch GFS model data on a 0.5&deg; scale, based on the request of staff forecasters. We fetch the data from [this server](https://www.ncei.noaa.gov/data/global-forecast-system/access/grid-004-0.5-degree/forecast/).
+
+##### Model Run Cycles
+
+NOAA runs the GFS model 4 times a day at 00:00, 06:00, 12:00, and 18:00 UTC. For each of these model runs, the model predicts weather at 3-hour intervals for 384 hours (16 days) in advance. Since Morecast users have indicated that they make forecasts up to a maximum of 10 days in advance, we only fetch GFS model data up to 240 hours (10 days) in advance.
+
+Additionally, GFS model data is only relevant to Morecast users for the 12:00 PST (13:00 PDT) timeframe - this is the time of day that a fire weather forecast is issued for. Noon PST corresponds to 20:00 UTC, but the GFS model has no output for this hour. We therefore fetch data for the 18:00 and 21:00 UTC hours, and subsequently perform linear interpolation to calculate approximate weather values for 20:00 UTC.
+
+The filename convention that NOAA uses when publishing GFS model data follows this pattern (as an example):
+
+> gfs_4_20230214_0600_003.grb2
+
+where '20230214' is the date that the model was run (in YYYYMMDD format), '0600' is the time at which the model was run (06:00 UTC), and '003' indicates that the weather prediction is for 003 hours after the model run time, which would be 09:00 UTC on Feb 14, 2023.
+
+Therefore, to predict weather for 12:00 PST, we must fetch the following model run cycles:
```

Comment:
I'm not sure I understand the why for requiring these model run cycles for 12:00PST when above it says:
> Noon PST corresponds to 20:00 UTC, but the GFS model has no output for this hour. We therefore fetch data for the 18:00 and 21:00 UTC hours, and subsequently perform linear interpolation to calculate approximate weather values for 20:00 UTC.

Reply:
I just spent some time trying to wrap my head around this as well. I think some clarification would be helpful.

I'm guessing GFS gives predictions for specific times like ' the predicted temperature at 18:00 UTC will be 5*C'. Rather, I think it gives predictions at intervals, eg. the temperature 3 hours from now will be 5*C, 6 hours from now 7*C, etc.'

Reply:
Added an extra sentence to the doc to explain why we use all 4 model run cycles.

Reply:
@dgboss  yes you are correct - it's intervals, not specific timestamps. Will try to explain that better in the doc.